sunpy/map/maputils.py

```diff
@@ -1,7 +1,267 @@
+
+
+def is_all_on_disk(smap):
+    """
+    Checks if the entire map is on the solar disk.
+
+    The check is performed by calculating the distance of every pixel
+    from the center of the Sun. If they are all on-disk, then the
+    function returns `True`. Otherwise, the function returns `False`.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A map in helioprojective Cartesian coordinates.
+
+    Returns
+    -------
+    `~bool`
+        Returns `True` if all map pixels are strictly less than one solar radius
+        away from the center of the Sun.
+    """
+    return np.all(coordinate_is_on_disk(all_coordinates_from_map(smap)))
+
+
+def contains_limb(smap):
+    """
+    Checks if a map contains a portion of the solar limb.
+
+    The check is performed by calculating the distance of every pixel from
+    the center of the Sun.  If at least one pixel is on disk and at least one
+    pixel is off disk, the function returns `True`. Otherwise, the function
+    returns `False`.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A map in helioprojective Cartesian coordinates.
+
+    Returns
+    -------
+    `~bool`
+        Returns `True` If at least one pixel is on disk and at least one pixel
+        is off disk.
+
+    Notes
+    -----
+    This will also return `True` if the entire solar
+    limb is within the field of view of the map.  Also in the case
+    of coronagraph images the limb itself need not be observed.
+    """
+    on_disk = coordinate_is_on_disk(all_coordinates_from_map(smap))
+    return np.logical_and(np.any(on_disk), np.any(~on_disk))
+
+
+def on_disk_bounding_coordinates(smap):
+    """
+    Returns the the bottom left and top-right coordinates of the smallest
+    rectangular region that contains all the on-disk pixels in the input map.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A map in helioprojective Cartesian coordinates.
+
+    Returns
+    -------
+    `~astropy.coordinates.SkyCoord`
+        A `~astropy.coordinates.SkyCoord` of length 2 such that the
+        first entry is the bottom left coordinate and the second entry is the
+        top right coordinate of the smallest rectangular region that contains
+        all the on-disk pixels in the input map.
+    """
+    # Check that the input map is not all off disk.
+    if is_all_off_disk(smap):
+        raise ValueError("The entire map is off disk.")
+
+    # Get all the coordinates from the input map
+    coordinates = all_coordinates_from_map(smap)
+
+    # Find which coordinates are on the disk
+    on_disk = coordinate_is_on_disk(coordinates)
+    on_disk_coordinates = coordinates[on_disk]
+
+    # The bottom left and top right coordinates that contain
+    # the on disk coordinates.
+    tx = on_disk_coordinates.Tx.value
+    ty = on_disk_coordinates.Ty.value
+    return SkyCoord([np.nanmin(tx), np.nanmax(tx)] * u.arcsec,
```

Comment:
Inconsistent API. A function above returns the same kind of result, a list of corners but here it is provided as an array as opposed to a dictionary as it was in `map_edges`. Why not return all of the corners as above or change `map_edges` to only return bottom left and top-right coordinate?

Reply:
`map_edges` does not return a list of corners, it returns all the pixels along the edge of the map. `map_edges` is needed in order to calculate a bounding box that contains the differentially rotated map - differentially rotating the corners is not sufficient.

Reply:
So there is no inconsistent API